api/src/pcapi/core/users/young_status.py

```diff
@@ -55,6 +64,18 @@ def young_status(user: models.User) -> YoungStatus:
         return Beneficiary()
 
     if user.eligibility is not None:
-        return Eligible()
+        if (
+            subscription_api.get_identity_check_subscription_status(user, user.eligibility)
+            == subscription_models.SubscriptionItemStatus.PENDING
+        ):
+            return Eligible(subscription_status=SubscriptionStatus.HAS_SUBSCRIPTION_PENDING)
+
+        if subscription_api.has_subscription_issues(user):
+            return Eligible(subscription_status=SubscriptionStatus.HAS_SUBSCRIPTION_ISSUES)
+
+        if subscription_api.get_next_subscription_step(user) is not None:
+            return Eligible(subscription_status=SubscriptionStatus.HAS_TO_COMPLETE_SUBSCRIPTION)
+
+        # should never happen
```

Comment:
On peut faire une étude sur la db pour voir s'il y a des cas ?
Les règles sont un peu bricolées donc on n'en est pas sûr :/ 
C'est vrai que c'est un peu dommage de ne pas avoir de fallback.
Ça devrait être amélioré par le ticket 17787

Reply:
> On peut faire une étude sur la db pour voir s'il y a des cas ?

aucune idée de comment faire ça =(

> Les règles sont un peu bricolées donc on n'en est pas sûr :/

est-ce que les tests de `young_status` te paraissent insuffisant ou faux d'un point de vue métier ?

est-ce qu'il y a des cas que tu imagines qui pourraient ne pas etre correctement gérés ?

> C'est vrai que c'est un peu dommage de ne pas avoir de fallback.

quel fallback imaginerais-tu ? (là, de fait, c'est non éligible)

> Ça devrait être amélioré par le ticket 17787

espérons !!

Reply:
> > On peut faire une étude sur la db pour voir s'il y a des cas ?
> 
> aucune idée de comment faire ça =(
tu peux faire un script python qui itère sur tous tes utilisateurs cibles et qui calcule tes 3 booleens -> à voir avec Corentin ?

> > Les règles sont un peu bricolées donc on n'en est pas sûr :/
> 
> est-ce que les tests de `young_status` te paraissent insuffisant ou faux d'un point de vue métier ?

cf mon message ci-dessus, ça me parait faux (wording, ordre de logique).

> est-ce qu'il y a des cas que tu imagines qui pourraient ne pas etre correctement gérés ?
> 
> > C'est vrai que c'est un peu dommage de ne pas avoir de fallback.
> 
> quel fallback imaginerais-tu ? (là, de fait, c'est non éligible)
> 
> > Ça devrait être amélioré par le ticket 17787
> 
> espérons !!

ma reco serait de faire le ticket 17787 d'abord

Reply:
en terme de priorité, il me semble que : 
* le 17787 sera fait dans quelques itérations
* cette US est bloquante pour la suite d'un chantier et on aimerait bien l'avoir pour la prochaine MeS (demain)